packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -877,6 +897,23 @@ export const stripeRoutes = (
       },
       handler: (request: AuthRequest) => stripeHandler.previewInvoice(request),
     },
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/invoice/preview-subsequent',
+      options: {
+        auth: false,
+        response: {
+          schema: invoiceDTO.subsequentInvoicePreviewSchema as any,
+        },
+        validate: {
+          payload: {
+            subscriptionId: validators.subscriptionsSubscriptionId.required(),
+          },
+        },
+      },
```

Comment:
Why one subscription at a time?  Would it be easier if we simply get all the previews for the customer?

Reply:
I left a [couple of replies](https://github.com/mozilla/fxa/pull/11890#discussion_r809500476) regarding the response object.

As to why it's a POST, I just copied the logic from previewInvoice. I'm guessing that it could be because of potential special characters in the request, so having it in the body is safer than in the query string?

As for local testing, it works for me. (See screenshot below)
![image](https://user-images.githubusercontent.com/10620585/154718534-201fbf03-c3b6-4258-8e58-8e9c7ab1f8da.png)

Reply:
Actually, now that I think about it, it'd probably be better to just pass in an array of subscriptionId's, since we already have that available on the front end. So the changes would be this.
- Keep as POST, update body to be array of subscriptionIds
- Add Auth
- Return array of results.

Reply:
> As to why it's a POST, I just copied the logic from previewInvoice. I'm guessing that it could be because of potential special characters in the request, so having it in the body is safer than in the query string?

We use GET for the /productname endpoint, passing in a product ID as a query string, and I would assume a subscription ID has the same character set as a product ID.

I asked Ben [in Slack](https://mozilla.slack.com/archives/CLV3KMZ8B/p1645219040634789) if he recalls, since he implemented that endpoint.

Reply:
Cool got it to work as a POST with the updated array response and authn.

![Screen Shot 2022-02-18 at 5 29 40 PM](https://user-images.githubusercontent.com/17437436/154770132-8fcbb9b9-c662-455c-ad17-65645e96cfbd.png)